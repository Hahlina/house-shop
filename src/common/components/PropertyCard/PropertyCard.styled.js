import styled from 'styled-components';

export const PropertyCardStyled = styled.div`
    overflow: hidden;
    width: 350px;
    height: 500px;
    border-radius: 26px;
    background: ${({ theme: { colors } }) => colors.white};
    border: 1px solid ${({ theme: { colors } }) => colors.gray};
    transition:
        transform 0.5s ease,
        box-shadow 0.3s ease;

    &:hover {
        transform: scale(1.02);
        box-shadow: 0 34px 36px 0 ${({ theme: { colors } }) => colors.black100};
    }

    img {
        height: 280px;
        width: 100%;
        object-fit: cover;
        object-position: center;
    }

    div.preview-information {
        display: flex;
        flex-direction: column;
        justify-content: space-between;
        padding: 25px;
        height: 200px;
    }

    p {
        font-size: 23px;
    }

    p.card-title {
        font-size: 25px;
        font-weight: 700;
    }

    p.price {
        color: ${({ theme: { colors } }) => colors.orange};
    }
`;
