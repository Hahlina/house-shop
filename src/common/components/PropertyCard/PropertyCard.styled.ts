import { Card, CardContent, CardMedia } from '@mui/material';
import styled from 'styled-components';

export const PropertyCardStyled = styled(Card)`
    width: 350px;
    height: 500px;
    border: 1px solid ${({ theme: { colors } }) => colors.gray};
    transition: box-shadow 0.3s ease;

    &:hover {
        box-shadow: 0 34px 36px 0 ${({ theme: { colors } }) => colors.black100};
    }
`;

export const CardMediaStyled = styled(CardMedia)`
    height: 280px;
`;

export const CardContentStyled = styled(CardContent)`
    display: flex;
    flex-direction: column;
    height: 220px;

    & > * {
        height: 33%;
    }

    p {
        font-size: 20px;
    }

    p.title {
        font-weight: 700;
    }

    p.price {
        color: ${({ theme: { colors } }) => colors.orange};
        font-size: 25px;
    }
`;
