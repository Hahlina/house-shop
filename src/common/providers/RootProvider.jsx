import { StyleProvider } from './StyleProviders';
import { StoreProvider } from './StoreProvider';

export const RootProvider = ({ children }) => (
    <StoreProvider>
        <StyleProvider>{children}</StyleProvider>
    </StoreProvider>
);
