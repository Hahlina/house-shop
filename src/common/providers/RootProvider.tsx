import type { FC, PropsWithChildren } from 'react';

import { StyleProvider } from './StyleProviders';
import { StoreProvider } from './StoreProvider';
import { MaterialProvider } from './MaterialProvider';

export const RootProvider: FC<PropsWithChildren> = ({ children }) => (
    <StoreProvider>
        <MaterialProvider>
            <StyleProvider>{children}</StyleProvider>
        </MaterialProvider>
    </StoreProvider>
);
