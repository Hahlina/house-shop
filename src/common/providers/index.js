export * from "./RootProvider";
export * from "./StyleProviders/StyleProviders";
