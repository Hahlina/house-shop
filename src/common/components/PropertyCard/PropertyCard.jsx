import { Link } from 'react-router-dom';

import { CardActionArea, Typography } from '@mui/material';

import { convertPrice } from 'common/utils';
import { APP_ROUTES } from 'common/constants';
import { PropertyCardStyled, CardMediaStyled, CardContentStyled } from './PropertyCard.styled';

export const PropertyCard = ({ address, preview_image, price, title, id }) => {
    const convertedPriceToUsd = convertPrice(price);

    return (
        <Link to={`${APP_ROUTES.PROPERTY}/${id}`}>
            <PropertyCardStyled>
                <CardActionArea>
                    <CardMediaStyled image={preview_image} />
                    <CardContentStyled>
                        <Typography className="title">{title}</Typography>
                        <Typography className="address">{address}</Typography>
                        <Typography className="price">{convertedPriceToUsd}</Typography>
                    </CardContentStyled>
                </CardActionArea>
            </PropertyCardStyled>
        </Link>
    );
};
