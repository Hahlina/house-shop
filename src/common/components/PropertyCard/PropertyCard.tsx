import type { FC } from 'react';
import { Link } from 'react-router-dom';

import { CardActionArea, Typography } from '@mui/material';

import { convertPrice } from 'common/utils';
import { APP_ROUTES } from 'common/constants';
import type { IProperty } from 'common/types';
import { PropertyCardStyled, CardMediaStyled, CardContentStyled } from './PropertyCard.styled';

interface IPropertyCard extends IProperty {}

export const PropertyCard: FC<IPropertyCard> = ({ address, preview_image, price, title, id }) => {
    const convertedPriceToUsd = convertPrice(price);

    return (
        <Link to={`${APP_ROUTES.PROPERTY}/${id}`}>
            <PropertyCardStyled>
                <CardActionArea>
                    <CardMediaStyled image={preview_image} />
                    <CardContentStyled>
                        <Typography className="title">{title}</Typography>
                        <Typography className="address">{address}</Typography>
                        <Typography className="price">{convertedPriceToUsd}</Typography>
                    </CardContentStyled>
                </CardActionArea>
            </PropertyCardStyled>
        </Link>
    );
};
