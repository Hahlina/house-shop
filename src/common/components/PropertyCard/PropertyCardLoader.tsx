import { Box, Skeleton } from '@mui/material';

import { CardContentStyled } from './PropertyCard.styled';

export const PropertyCardLoader = () => (
    <Box height={500} width={350}>
        <Skeleton animation="wave" height={250} sx={{ transform: 'scale(1, 1)' }} />
        <CardContentStyled>
            <Skeleton animation="wave" height={50} />
            <Skeleton animation="wave" height={50} />
            <Skeleton animation="wave" height={50} />
        </CardContentStyled>
    </Box>
);
