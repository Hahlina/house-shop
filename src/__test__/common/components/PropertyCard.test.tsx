import { render, screen } from '@testing-library/react';

import { PropertyCard } from 'common/components/PropertyCard/PropertyCard';
import { RootProvider } from 'common/providers/RootProvider';

const property = {
    address: '123 Main St',
    preview_image: 'image.jpg',
    price: 100000,
    title: 'Beautiful House',
    id: '123'
};

jest.mock('react-router-dom', () => ({
    Link: 'a'
}));

describe('PropertyCard', () => {
    it('should render a title', () => {
        render(<PropertyCard {...property} />, {
            wrapper: RootProvider
        });
        expect(screen.getByText(property.title)).toBeInTheDocument();
    });

    it("should display the property's preview image", () => {
        render(<PropertyCard {...property} />, {
            wrapper: RootProvider
        });
        expect(screen.getByRole('img')).toHaveStyle(`background-image: url(${property.preview_image})`);
    });

    it('should match snapshot', () => {
        const { container } = render(<PropertyCard {...property} />, {
            wrapper: RootProvider
        });
        expect(container).toMatchSnapshot();
    });
});
