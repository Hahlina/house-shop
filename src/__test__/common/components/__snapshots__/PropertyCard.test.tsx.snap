// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard should match snapshot 1`] = `
.c0 {
  width: 350px;
  height: 500px;
  border: 1px solid #979797;
  transition: box-shadow 0.3s ease;
}

.c0:hover {
  box-shadow: 0 34px 36px 0 #00000021;
}

.c1 {
  height: 280px;
}

.c2 {
  display: flex;
  flex-direction: column;
  height: 220px;
}

.c2>* {
  height: 33%;
}

.c2 p {
  font-size: 20px;
}

.c2 p.title {
  font-weight: 700;
}

.c2 p.price {
  color: #F4511E;
  font-size: 25px;
}

<div>
  <a
    to="property/123"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root c0 css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardMedia-root c1 css-16cvb1l-MuiCardMedia-root"
          role="img"
          style="background-image: url(image.jpg);"
        />
        <div
          class="MuiCardContent-root c2 css-46bh2p-MuiCardContent-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 title css-ahj2mt-MuiTypography-root"
          >
            Beautiful House
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 address css-ahj2mt-MuiTypography-root"
          >
            123 Main St
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 price css-ahj2mt-MuiTypography-root"
          >
            $100,000.00
          </p>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </a>
</div>
`;
